src/app/components/common/datepicker/datepicker.component.html

```diff
@@ -0,0 +1,22 @@
+<div data-module="ds-datepicker" class="ds_datepicker">
+  <label class="ds_label" [for]="inputId">{{ label }}</label>
+  <p class="ds_hint-text">{{ hint }}</p>
+  <!-- Error message if date invalid -->
+  <div *ngIf="areDateErrors()" [id]="inputId + '-errors'">
+    <p *ngFor="let error of errors" class="govuk-error-message">
```

Comment:
Consider using the latest control flow Angular v17 syntax here: https://angular.dev/essentials/conditionals-and-loops#if-block